cardinal/message/message.go

```diff
@@ -31,7 +31,7 @@ type MessageType[In, Out any] struct { //nolint:revive // this is fine for now.
 	outEVMType *ethereumAbi.Type
 }
 
-func validateIfStruct[T any]() bool {
+func IsStruct[T any]() bool {
```

Comment:
eh not really a fan of making something public just to test it. i'd say either:

- make a unit test file to test this directly
- indirectly test it via public method (can probs just add on this test: `TestReadTypeNotStructs` (not sure why that test has the wrong name actually) to ensure it works on both pointers to structs and non-ptr structs)

Reply:
I'm not sure what you mean by make a unit_test to test this directly. But I went with that route because it seems the most straightforward. 

I made a file called isstruct_test.go that's part of the same message package. Let me know if that fits into what you were looking for.

Reply:
We talked about it on slack and agreed to go with the second route here.